client-src/elements/chromedash-gate-column.js

```diff
@@ -0,0 +1,255 @@
+import {LitElement, css, html, nothing} from 'lit';
+import './chromedash-activity-log';
+import {showToastMessage, findProcessStage} from './utils.js';
+
+import {SHARED_STYLES} from '../sass/shared-css.js';
+
+
+export const STATE_NAMES = [
+  [7, 'No response'],
+  [1, 'N/a or Ack'],
+  [2, 'Review requested'],
+  [3, 'Review started'],
+  [4, 'Needs work'],
+  [5, 'Approved'],
+  [6, 'Denied'],
+];
+
+
+class ChromedashGateColumn extends LitElement {
+  static get styles() {
+    return [
+      ...SHARED_STYLES,
+      css`
+       #votes-area {
+         margin: var(--content-padding) 0;
+       }
+
+       #votes-area table {
+         border-spacing: var(--content-padding-half) var(--content-padding);;
+       }
+
+       #votes-area th {
+         font-weight: bold;
+       }
+
+       #review-status-area {
+         margin: var(--content-padding-half) 0;
+       }
+
+    `];
+  }
+
+  static get properties() {
+    return {
+      user: {type: Object},
+      feature: {type: Object},
+      stage: {type: Object},
+      gate: {type: Object},
+      process: {type: Object},
+      votes: {type: Array},
+      comments: {type: Array},
+      loading: {type: Boolean},
+      needsSave: {type: Boolean},
+    };
+  }
+
+  constructor() {
+    super();
+    this.user = {};
+    this.feature = {};
+    this.stage = {};
+    this.gate = {};
+    this.process = {};
+    this.votes = [];
+    this.comments = [];
+    this.loading = true;
+    this.needsSave = false;
+  }
+
+  setContext(feature, stage, gate) {
+    this.loading = true;
+    this.feature = feature;
+    this.stage = stage;
+    this.gate = gate;
+    const featureId = this.feature.id;
+    Promise.all([
+      window.csClient.getFeatureProcess(featureId),
+      window.csClient.getApprovals(featureId),
+      window.csClient.getComments(featureId),
+    ]).then(([process, approvalRes, commentRes]) => {
+      this.process = process;
+      this.votes = approvalRes.approvals.filter((v) =>
+        v.gate_id == this.gate.id);
+      this.comments = commentRes.comments;
+      this.needsSave = false;
+      this.loading = false;
+    }).catch(() => {
+      showToastMessage('Some errors occurred. Please refresh the page or try again later.');
+      this.handleCancel();
```

Comment:
where is handleCancel defined?

Reply:
It was not defined yet.  I have added it now.